qview: migrate to `Qt6` and add `HEIF` support

magicquark opened the pull request:
### Migrate to `Qt6` and add `HEIF` support to `qView`

#### Migrate to Qt6

- `qView` has supported `Qt6` since version 5.0, see the [changelog](https://github.com/jurplel/qView/releases/tag/5.0).

- Migrating to `Qt6` allows the use of the `Qt6` library `kdePackages.kimageformats`, which adds support, among other formats, for `HEIF` to `qView`.

- Qt X11 Extras was removed in `Qt6`, see [Qt6 docs](https://doc.qt.io/qt-6/extras-changes-qt6.html#changes-to-qt-x11-extras).

#### Add Support for `HEIF`
- Upstream `qView` supports `HEIF` formats (e.g., `HEIC` used by iPhones), but `qView` fails to open these images with the error:

> "Error occurred opening "IMG_XXXX.HEIC": Unsupported image format (Error 3)".

- The addition of `kdePackages.kimageformats` adds support for `HEIF` fixing the error.

## Things done
- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

magicquark commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 453428`
Commit: [`4a0678afc721a7ae0345c30846a0b9da1471b34b`](https://github.com/NixOS/nixpkgs/commit/4a0678afc721a7ae0345c30846a0b9da1471b34b) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/4a0678afc721a7ae0345c30846a0b9da1471b34b..pull/453428/head))
Merge: [`462ee60d68379af55ec20956aeab89d61439e028`](https://github.com/NixOS/nixpkgs/commit/462ee60d68379af55ec20956aeab89d61439e028)

Logs: https://github.com/magicquark/nixpkgs-review-gha/actions/runs/18625864037


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>libsForQt5.qt-heif-image-plugin (plasma5Packages.qt-heif-image-plugin)</li>
    <li>qview</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>libsForQt5.qt-heif-image-plugin (plasma5Packages.qt-heif-image-plugin)</li>
    <li>qview</li>
  </ul>
</details>

---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>libsForQt5.qt-heif-image-plugin (plasma5Packages.qt-heif-image-plugin)</li>
    <li>qview</li>
  </ul>
</details>

---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>libsForQt5.qt-heif-image-plugin (plasma5Packages.qt-heif-image-plugin)</li>
    <li>qview</li>
  </ul>
</details>

magicquark commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 453428`
Commit: [`add92bfd7147771cd7c190afa8699ab90794ac1b`](https://github.com/NixOS/nixpkgs/commit/add92bfd7147771cd7c190afa8699ab90794ac1b) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/add92bfd7147771cd7c190afa8699ab90794ac1b..pull/453428/head))
Merge: [`c88068446bca147eca349681e5fa9974d3ac8b58`](https://github.com/NixOS/nixpkgs/commit/c88068446bca147eca349681e5fa9974d3ac8b58)

Logs: https://github.com/magicquark/nixpkgs-review-gha/actions/runs/18628128566


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>libsForQt5.qt-heif-image-plugin (plasma5Packages.qt-heif-image-plugin)</li>
    <li>qview</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>libsForQt5.qt-heif-image-plugin (plasma5Packages.qt-heif-image-plugin)</li>
    <li>qview</li>
  </ul>
</details>

---
### `x86_64-darwin` (sandbox = relaxed)
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>libsForQt5.qt-heif-image-plugin (plasma5Packages.qt-heif-image-plugin)</li>
    <li>qview</li>
  </ul>
</details>

---
### `aarch64-darwin` (sandbox = relaxed)
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>libsForQt5.qt-heif-image-plugin (plasma5Packages.qt-heif-image-plugin)</li>
    <li>qview</li>
  </ul>
</details>

K900 commented:
`kdePackages.kimageformats` includes a HEIF plugin for Qt 6. I'd rather not see any more new Qt5 stuff added.

magicquark committed:
qview: migrate to qt6 and add HEIF support

- `qview` has supported Qt6 since version 5.0; this commit migrates
  fully to it.

- Qt6 enables the use of `kdePackages.kimageformats`, which provides
  support for 'HEIF' among other image formats.

magicquark committed:
qview: modernize

magicquark commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 453428`
Commit: [`305254dc6f446d9e634f51abd7e9419f37c68b91`](https://github.com/NixOS/nixpkgs/commit/305254dc6f446d9e634f51abd7e9419f37c68b91) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/305254dc6f446d9e634f51abd7e9419f37c68b91..pull/453428/head))
Merge: [`435f5d112a79dd87c5ee7ea6c4c5132b5fc9b170`](https://github.com/NixOS/nixpkgs/commit/435f5d112a79dd87c5ee7ea6c4c5132b5fc9b170)

Logs: https://github.com/magicquark/nixpkgs-review-gha/actions/runs/18666883182


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>qview</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>qview</li>
  </ul>
</details>

---
### `x86_64-darwin`
:white_check_mark: *No rebuilds*

---
### `aarch64-darwin`
:white_check_mark: *No rebuilds*

K900 commented:
Awesome, thanks.
